apps/ui/src/views/Swap/SwapList.tsx

```diff
@@ -75,9 +86,32 @@ export const SwapList: React.FC = () => {
 
   const { pendingOrders, historyOrders } = useSwapOrders(orderList);
   const [orderSelectorStatus, setOrderSelectorStatus] = useState(OrderSelectorStatus.Pending);
+  const [currentPairOnly, setCurrentPairOnly] = useState(false);
+  const pairFilter = useCallback(
+    (order: SwapOrder) => {
+      const currentPair = [tokenA, tokenB];
+      const { amountIn, amountOut } = order;
+      const pair = [
+        EthModel.isShadowEthAsset(amountIn) && order.type === SwapOrderType.CrossChainOrder
+          ? amountIn.shadowFrom
+          : amountIn,
+        EthModel.isShadowEthAsset(amountOut) && order.type === SwapOrderType.CrossChainOrder
+          ? amountOut.shadowFrom
+          : amountOut,
+      ];
+      return (
+        differenceWith(currentPair, pair, (a, b) => {
+          const model = Models.get(a.chainType)!;
```

Comment:
Although `a.chainType` mostly has matching `ChainType`, `!` symbol it's better not to put it in the code

```ts
const model = Models.get(a.chainType)
if(!model) return false;
return model.isCurrentChainAsset(a) && model.isCurrentChainAsset(b) && model.equals(a, b);
```